pkg/cmd/experimental/syncgroups/groupsyncer.go

```diff
@@ -0,0 +1,143 @@
+	if err != nil {
+		errors = append(errors, err)
+		return errors
+	}
+
+	for _, ldapGroupUID := range ldapGroupUIDs {
+		// get membership data
+		memberEntries, err := s.GroupMemberExtractor.ExtractMembers(ldapGroupUID)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		// determine OpenShift Users' usernames for LDAP group members
+		usernames, err := s.determineUsernames(memberEntries)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		// update the OpenShift Group corresponding to this record
+		err = s.updateGroup(ldapGroupUID, usernames)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+	}
+	return errors
+}
+
+// determineUsers determines the OpenShift Users that correspond to a list of LDAP member entries
+func (s *LDAPGroupSyncer) determineUsernames(members []*ldap.Entry) ([]string, error) {
+	var usernames []string
+	for _, member := range members {
+		username, err := s.UserNameMapper.UserNameFor(member)
+		if err != nil {
+			return nil, err
+		}
+		usernames = append(usernames, username)
+	}
+	return usernames, nil
+}
+
+// updateGroup finds or creates the OpenShift Group that needs to be updated, updates its' data, then
+// uses the GroupClient to update the Group record
+func (s *LDAPGroupSyncer) updateGroup(ldapGroupUID string, usernames []string) error {
+	// find OpenShift Group to update
+	group, err := s.findGroup(ldapGroupUID)
+	if err != nil {
+		return err
+	}
+
+	// overwrite Group Users data
+	group.Users = usernames
+
+	// add LDAP-sync-specific annotations
+	group.Annotations[syncutil.LDAPUID] = ldapGroupUID
+	group.Annotations[syncutil.LDAPSyncTime] = ISO8601(time.Now())
+
+	// add LDAP URL label
+	group.Labels[syncutil.LDAPURL] = s.Host
+
+	_, err = s.GroupClient.Update(group)
+	return err
+}
+
+// findGroup finds the OpenShift Group for the LDAP group UID and ensures that the OpenShift Group found
+// was created as a result of a previous LDAP sync from the same LDAP group.
+func (s *LDAPGroupSyncer) findGroup(ldapGroupUID string) (*ouserapi.Group, error) {
+	groupName, err := s.GroupNameMapper.GroupNameFor(ldapGroupUID)
+	if err != nil {
+		return nil, err
+	}
+
+	group, err := s.GroupClient.Get(groupName)
+	if err != nil {
+		return nil, fmt.Errorf("could not get group for name: %s", groupName)
+	}
+
+	url, exists := group.Labels[syncutil.LDAPURL]
+	if !exists || url != s.Host {
+		return nil, fmt.Errorf("group %s's %s label did not match sync host: wanted %s, got %s",
+			group.Name, syncutil.LDAPURL, s.Host, url)
+	}
+	uid, exists := group.Annotations[syncutil.LDAPUID]
+	if !exists || uid != ldapGroupUID {
+		return nil, fmt.Errorf("group %s's %s annotation did not match sync host: wanted %s, got %s",
+			group.Name, syncutil.LDAPUID, ldapGroupUID, uid)
+	}
+	return group, nil
+}
+
+// ISO8601 returns an ISO 6801 formatted string from a time.
+func ISO8601(t time.Time) string {
+	var tz string
+	if zone, offset := t.Zone(); zone == "UTC" {
```

Comment:
Given that the code implements the standard correctly, what would you like me to change? From Zulu to `0` offset?